### ISSUE TITLE
React hooks & context

### ISSUE THREAD

**Issue by konsumer** [Yomguithereal/react-blessed#88]
I'd love to be able to use [react hooks](https://reactjs.org/hooks) in react-blessed.

It's an RFC, now, but looks like it will probably solidify into mainline react, soon. In order to test, I installed `react@16.7.0-alpha.2`.

As it is, I get this error:

```
Invariant Violation: Hooks can only be called inside the body of a function component.
```

with this code:

```js
import React, { useState, useEffect } from 'react'
import fetch from 'node-fetch'

function App () {
  const [posts, setPosts] = useState([])
  useEffect(async () => {
    const items = await fetch(
      'https://jsonplaceholder.typicode.com/posts'
    ).then(r => r.json())
    setPosts(items.map(i => i.title))
  })

  return (
    <list items={posts} />
  )
}
```

To verify that the hooks work, etc, [here](https://codesandbox.io/s/py6r7xq6z0) is a similar thing in a sandbox, in react-dom.

**Comment by konsumer** [comment]
[here](https://github.com/konsumer/react-blessed-demo) is a quick demo of the code above, so you can try it out quickly.

**Comment by Yomguithereal** [comment]
Hello @konsumer. I agree it would be nice to have hooks support. Unfortunately I am not even sure the current codebase is able to work with react@16 since a lot of changes were made to the reconciler logic. I have not much time to work on this but will of course consider any PR.

**Comment by konsumer** [comment]
I noticed similar with [context](https://reactjs.org/docs/context.html) (in trying to work around no hooks) and was about to make another issue. :) 

I'd be happy to make a PR, but I'm not really sure where to start. It looks like you are using [react-reconciler](https://www.npmjs.com/package/react-reconciler) with fibers, which is basically how I would do it, and as far as I can tell all the code is fine, and should work great. Maybe it just needs new versions of a few things. I will experiment with it, and report back.

**Comment by konsumer** [comment]
Ok, I filled in [the demo project](https://github.com/konsumer/react-blessed-demo) a bit more to test hooks & context, including demos that work in react-dom, and will change the title of the isse to include both. I'll start looking at what needs to be added to react-blessed to make it work.

**Comment by konsumer** [comment]
I got a good start, I think. Supports `@next` of react, and new fibber stuff.

Maybe I am doing it wrong, but I'm not sure how to make buttons `onPress` to work, and it's keeping demos form working (button-presses trigger state/context changes.) Any help would be appreciated. I all I needed to do was this:

```js
<button onPress={a => console.error('press', a)}>TEST</button>
```

but it doesn't seem to be calling `onPress`.
